osprey/utils.py

```diff
@@ -34,6 +36,38 @@ def replace_filenames(config, temp_config):
     temp_config.writelines(newdata)
 
 
+def replace_urls(config, outdir):
+    """
+    Copy https URLs to local storage and replace URLs
+    with local paths in config file.
+    Parameters:
+        config (str): Config file
+        outdir (str): Output directory
+    """
+    read_config = open(config, "r")
+    filedata = read_config.readlines()
+    read_config.close()
+
+    for i in range(len(filedata)):
+        if "https" in filedata[i]:
+            url = filedata[i].split(" ")[-1]  # https url is last word in line
+            url = url.rstrip()  # remove \n character at end
+            r = requests.get(url)
+            filename = url.split("/")[-1]
+            prefix, suffix = filename.split(".")
+            suffix = "." + suffix
+            local_file = NamedTemporaryFile(
+                suffix=suffix, prefix=prefix, dir=outdir, delete=False
+            )
+            local_file.write(r.content)
+            filedata[i] = filedata[i].replace(url, local_file.name)
```

Comment:
While this is _fine_, we generally don't want to use indices if we can help it. It is more understandable to iterate through the iterable object : `for data in filedata:...`.

Reply:
I used the index because I wasn't sure if the assignment in line 63 would modify the `filedata` list if I iterated through each element.